052-record-reconciled-versions.md

```diff
@@ -0,0 +1,190 @@
+  Updating the status with the reconciled kafka versions:
+  ```
+  status:
+    kafkaVersion: 3.5.1
+  ```
+- Kafka Reconciler continues to install other components such as entity operator.
+  Once all components are deployed and updated, update `Kafka` CR with
+  ```
+  status:
+    operatorLastSuccessfulVersion: 0.38.0
+    kafkaVersion: 3.5.1
+  ```
+- User can now verify `Kafka.status.operatorLastSuccessfulVersion` and `Kafka.status.kafkaVersion` to check if it has reconciled to correct/expected levels.
+- if there is an error at any point prior to the Kafka Broker rollout `Kafka.status.kafkaVersion` will not be updated.
+- if there is an error at any point during reconcile `Kafka.status.operatorLastSuccessfulVersion` will not be updated.
+
+
+### Upgrade (with new mechanism)
+
+This defines an example where it is imagined that `0.37.0` had this mechanism already implemented, and is now upgrading to a new version `0.38.0` (with kafka upgrading from `3.5.0` to `3.5.1`)
+
+- On an upgrade, a user has a Kafka CR:
+  ```
+  status:
+    operatorLastSuccessfulVersion: 0.37.0
+    kafkaVersion: 3.5.0
+  ```
+- The KafkaReconciler reconciles the CR and creates `StrimziPodSet` CRs
+- StrimziPodSet reconciler reconciles the `SPS`s creating the Kafka Brokers
+- Kafka Reconciler picks up that the Kafka Brokers managed by the SPS are now Ready and finished installing.
+  Updating the status with the reconciled kafka versions:
+  ```
+  status:
+    operatorLastSuccessfulVersion: 0.37.0
+    kafkaVersion: 3.5.1
+  ```
+- Kafka Reconciler continues to install other components such as entity operator.
+  Once all components are deployed and updated, update `Kafka` CR with
+  ```
+  status:
+    operatorLastSuccessfulVersion: 0.38.0
+    kafkaVersion: 3.5.1
+  ```
+- User can now verify `Kafka.status.operatorLastSuccessfulVersion` and `Kafka.status.kafkaVersion` to check if it has reconciled to correct/expected levels.
+- if there is an error at any point prior to the Kafka Broker rollout `Kafka.status.kafkaVersion` will not be updated.
+- if there is an error at any point during reconcile `Kafka.status.operatorLastSuccessfulVersion` will not be updated.
+
+## `KafkaNodePool` considerations
+
+It might be that in the future each `KafkaNodePool` CR supports having its own Kafka version which might differ from other `KafkaNodePools` in the same Kafka cluster.
+This proposal should work with this method of upgrade, since the Kafka version + readiness checks will be track by the `StrimziPodSet`, so that when a user had finished upgrading via this method, the new `status.kafkaVersion` should be updated accordingly.
+In the potential case where a user is using multiple `KafkaNodePool`s for a cluster using different Kafka Versions, it is my recommendation that `status.kafkaVersion` is a comma delimitered (sequentially ordered) list of all these Kafka Versions rather than the minimum or maximum kafka version.
```

Comment:
```suggestion
In the potential case where a user is using multiple `KafkaNodePool`s for a cluster using different Kafka versions, it is my recommendation that `status.kafkaVersion` is a comma delimitered (sequentially ordered) list of all these Kafka Versions rather than the minimum or maximum Kafka version.
```